src/OrchardCore.Cms.Web/appsettings.json

```diff
@@ -79,6 +79,35 @@
     //  "LogoffPath": "Users/LogOff",
     //  "ChangePasswordUrl": "ChangePassword",
     //  "ExternalLoginsUrl": "ExternalLogins"
+    //},
+    //"OrchardCore_AutoSetup": {
+    //  "TriggerSetupUrl": "",
+    //  "RootTenant": {
+    //    "SiteName": "AutoSetup Example",
+    //    "SiteTimeZone": "Europe/Amsterdam",
+    //    "AdminUsername": "Admin",
+    //    "AdminEmail": "info@orchardproject.net",
+    //    "AdminPassword": "OrchardCoreRules1!",
+    //    "DatabaseProvider": "Sqlite",
+    //    "DatabaseConnectionString": "",
+    //    "DatabaseTablePrefix": "",
+    //    "RecipeName": "Agency"
+    //  },
+    //  "SubTenants": [
```

Comment:
I think this could just be one array, where you find the `Default` shell by it's name.

Reply:
I also thought about this, then decided to separate it explicitly since 
- Not everyone knows the Default naming convention
- The Default Tenant does not need `RequestUrlHost` and `RequestUrlPrefix` params

Reply:
I think it should use the default sections, including for other tenants.
I also think it should only initialize the tenant the first time it's requested, and not on the first request of the default one.

Reply:
@sebastienros sorry it is not clear to me, can you please rephrase your concern and/or give me an example?

Reply:
Not sure which part you didn't understand. What I mean is that last time I saw the code, the tenants were all initialized when the application is starting. It would be better if a tenant was auto-initialized on the first request to it. So if you open the default tenant, it's initialized, then if you open tenant1 it get initialized.

Reply:
I've implemented on-demand initialization please take a look